python/lsst/ap/association/trailedSourceFilter.py

```diff
@@ -0,0 +1,106 @@
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.utils.timer import timeMethod
+
+
+class TrailedSourceFilterConfig(pexConfig.Config):
+    """Config class for TrailedSourceFilterTask.
+    """
+    maxTrailLength = pexConfig.Field(
+        dtype=float,
+        doc='Maximum trail length permitted is less than 10 degrees/day. '
+            'This is a rate of 0.416 arcseconds per second. As trail length'
+            ' is measured in arcseconds, it is dependant on the length of '
+            'the exposure.',
+        default=0.416,
+    )
+
+
+class TrailedSourceFilterTask(pipeBase.Task):
+    """Find trailed sources in DIASources. This task checks the length of
+    trailLength in the DIASource catalog using a given
+     arcsecond/second rate from maxTrailLength and the exposure time. The
+     two values are used to calculate the maximum allowed trail length and
+     filters out any trail longer than the maximum.
+
+    """
+
+    ConfigClass = TrailedSourceFilterConfig
+    _DefaultName = "trailedSourceFilter"
+
+    @timeMethod
+    def run(self, dia_sources, diffIm):
+        """Find trailed sources which do not meet requirements and
+        will not be included in the diaSource catalog.
+
+        Parameters
+        ----------
+        dia_sources : `pandas.DataFrame`
+            New DIASources to be checked for trailed sources.
+        diffIm : `lsst.afw.image.ExposureF`
+            Difference image on which the DIASources were detected.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results struct with components.
+
+            - ``dia_sources`` : DIASource table that is free from unwanted
+             trailed sources. (`pandas.DataFrame`)
+
+            - ``trailed_dia_sources`` : DIASources that have trailed more
+            than 0.416 arcseconds/second*exposure_time. (`pandas.DataFrame`)
+        """
+
+        trail_mask = self.check_dia_source_trail(dia_sources, diffIm)
+
+        return pipeBase.Struct(
+            diaSources=dia_sources[~trail_mask].reset_index(drop=True),
+            trailedDiaSources=dia_sources[trail_mask].reset_index(drop=True))
+
+    def check_dia_source_trail(self, dia_sources, diffIm):
+        """Find DiaSources that have long trails.
+
+        Creates a mask for sources with lengths greater than 0.416
+        arcseconds/second multiplied by the exposure time.
+
+        Parameters
+        ----------
+        dia_sources : `pandas.DataFrame`
+            Input DIASources to check for trail lengths.
+        diffIm : `pandas.DataFrame`
+            Difference image on which the DIASources were detected.
+
+        Returns
+        -------
+        trail_mask : `pandas.DataFrame`
+            Boolean mask for DIASources which are greater than the
+            cutoff length.
+        """
+        diffIm_time = diffIm.getInfo().getVisitInfo().getExposureTime()
```

Comment:
Difference images come from visits. Given we can take 2 snaps and there is a gap between the snaps, is the relevant time here the exposure time of a snap or the duration of the visit including the gap (which is greater than the exposure time)?

(also, if `diffim` is an exposure then the docstring for `diffIm` is wrong).

Reply:
Is there a policy and/or actual code that defines what the "exposure time" of a snap-combined image is? [The docs for `VisitInfo`](http://doxygen.lsst.codes/stack/doxygen/x_mainDoxyDoc/classlsst_1_1afw_1_1image_1_1_visit_info.html#ab1318e25db6b09d117a8dc78311bff5c) just say:
> get exposure duration (shutter open time); (sec)

Since we can't yet run `CharacterizeImageTask` or `CalibrateImageTask` on multi-snap visits, I'm pretty sure that this case is untestable either way.

Reply:
At the moment it's defined to be the sum of the exposure time of the snaps. Unlike ObservationInfo, VisitInfo doesn't record the start and end time, only the midpoint (which for two snaps might be a time where no data are being taken).